src/base/bittorrent/session.cpp

```diff
@@ -1619,10 +1620,34 @@ void Session::loadLTSettings(lt::settings_pack &settingsPack)
 
     settingsPack.set_int(lt::settings_pack::max_queued_disk_bytes, diskQueueSize());
 
-    lt::settings_pack::io_buffer_mode_t mode = useOSCache() ? lt::settings_pack::enable_os_cache
-                                                              : lt::settings_pack::disable_os_cache;
-    settingsPack.set_int(lt::settings_pack::disk_io_read_mode, mode);
-    settingsPack.set_int(lt::settings_pack::disk_io_write_mode, mode);
+    switch (diskIOReadMode())
+    {
+    case DiskIOReadMode::DisableOSCache:
+        settingsPack.set_int(lt::settings_pack::disk_io_read_mode, lt::settings_pack::disable_os_cache);
+        break;
+    case DiskIOReadMode::EnableOSCache:
+    default:
+        settingsPack.set_int(lt::settings_pack::disk_io_read_mode, lt::settings_pack::enable_os_cache);
+        break;
+    }
+
+    switch (diskIOWriteMode())
+    {
+    case DiskIOWriteMode::DisableOSCache:
+        settingsPack.set_int(lt::settings_pack::disk_io_write_mode, lt::settings_pack::disable_os_cache);
+        break;
+    case DiskIOWriteMode::EnableOSCache:
+    default:
+        settingsPack.set_int(lt::settings_pack::disk_io_write_mode, lt::settings_pack::enable_os_cache);
+        break;
+    case DiskIOWriteMode::WriteThrough:
+#ifdef QBT_USES_LIBTORRENT2
+        settingsPack.set_int(lt::settings_pack::disk_io_write_mode, lt::settings_pack::write_through);
+#else
+        settingsPack.set_int(lt::settings_pack::disk_io_write_mode, lt::settings_pack::disable_os_cache);
+#endif
+        break;
```

Comment:
```suggestion
#ifdef QBT_USES_LIBTORRENT2
    case DiskIOWriteMode::WriteThrough:
        settingsPack.set_int(lt::settings_pack::disk_io_write_mode, lt::settings_pack::write_through);
        break;
#endif
```

Reply:
what will happen when someone selects writethrough from WebUI or when they downgrade.

Reply:
>what will happen when someone selects writethrough from WebUI or when they downgrade.

Same as https://github.com/qbittorrent/qBittorrent/pull/17005#discussion_r916769794. Additionally it will fall into `switch()` `default:` case.

Reply:
Done.